src/stampit.js

```diff
@@ -124,6 +132,16 @@ function compose(...stamps) {
     stamps.push(this);
   }
 
+  /*
+   * If stamp has unique displayName and is cached, return it. A composed stamp
+   * should have a unique displayName. Since the compose method overrides with last-in
+   * priority, the displayName should be at the end.
+   */
+  let displayName = stamps[stamps.length - 1].displayName || 'ReactStamp';
```

Comment:
Should you walk the stamps backward until you find a `displayName` or finish checking stamps?

I wouldn't worry about a performance hit, since there are typically only a handful of composed stamps...